resources/contents/blog/en/readme-driven-development.md

```diff
@@ -0,0 +1,75 @@
+---
+id: 201811111
+title: "The power of Readme Driven Development"
+date: 2018-11-11T16:55:22+02:00
+author: Rathes Sachchithananthan
+template: post
+description: Readme Driven Development (RDD) is a powerful approach in software development to write libraries that people can use and want to use
+image: /images/blog/readme-driven-development.jpg
+categories:
+  - Web
+tags:
+  - documentation
+  - concepts
+locale: en_US
+---
+
+A good piece of software has to follow one important rule: It has to solve the right problem and it has to solve the problem right. This rule works for big enterprise applications as well as for small libraries that serve as part of a bigger ecosystem.
+
+What it actually means is that as a software developer you have to create software that actually solves the problem that was identified and that the software has to be created in a way that others can actually use it. It does not make any sense to create the perfect, bug-free piece of software if nobody needs it. And at the same time it does not make any sense to create software that people just can't use because it is terribly (or even worse not at all) documented.
+
+## Introducing RDD
+
+There is a simple pattern to solve the latter problem of creating something that nobody can't use. It is that simple that it can be describe in one sentence, but at the same time there are a few misconceptions that it needs a whole article to clarify things:
+
+**Write the Readme first.**
+
+Read it again. Write the Readme first. With first, it means before anything. It means before you the first line of code, it means before the first test and it even means before you setup the project.
+
+And read it again. It says Readme. It does not say documentation, it does not say manual and it does not say specifications. Just Readme.
+
+**Write the Readme first.**
+
+That's all. This rule will help to avoid that pieces of software becomes useless once it is finished.
+
+## Implementing Readme Driven Development
+
+But what does it exactly mean to write the Readme first? It is pretty straight forward. Just before you start working, take the time and start writing a simple file that explains what it is all about.
+
+Start with a title and a subtitle that makes it directly clear what the topic is. Then continue with a small paragraph explaining what the code you write intends to do and what it is going to solve.
+
+Then follow with some *Getting Started* information, so that the user of this piece of software will be able to set things up and configure things if needed.
```

Comment:
```suggestion
Then follow with some *Getting Started* information, so that the user of this piece of software will be able to set things up and configure them if needed.
```